labs/lab-2-javascript-game/elicecilgame.js

```diff
@@ -0,0 +1,59 @@
+var grantHealth = 10;
+var userHealth = 40; 
+var winCount = 0;
+
+// class User {
+//     constructor()
+// }
+
+const startGame = function() {
+    var start = prompt("Would you like to fight the Champion?");
+    if(start === "yes") {
+        name = prompt("Tell us your name.");
+    }
+    startCombat();
+}
+
+const startCombat = function() {   
+    const damageRoll = function() {
+        var damage = Math.round(Math.random() + 1);
+        return damage;
+    }
+    const grantHit = function() {
+        grantHealth = (grantHealth - damageRoll());
+        console.log("Grant is now at", grantHealth, "HP!");  
+    }
+    const userHit = function() {
+        userHealth = (userHealth - damageRoll());
+        console.log(name, "is now at", userHealth, "HP!");
+    }
+    const wins = function() {
+        if (grantHealth === 0) {
+            winCount = (winCount + 1);
+            console.log(name, "has", winCount, "wins!");
+        }
+    }
+    const loss = function() {
+        if (userHealth === 0) {
+            console.log(name, "has been defeated!");
+        }
+    }
+    const oneDeath = function() {
+        while((grantHealth > 0) && (userHealth > 0 )) {
+            userHit();
+            grantHit();
+        }
+    }
```

Comment:
I effing love this list of functions. This is SO GOOD. Each function does one simple thing. Reading each function it's very easy to *believe* in it -- there's not much room for a bug to hide in any of these, because each one is straightforward. 

I'd tweak some of the names, slightly: `hitUser` instead of `userHit` -- if it's an action function, give it a verb-y name.

and `checkForWin` or `reportIfWon` or something like that instead of `wins`. 

Ditto for `grantHit` and `loss`.